Cargo.toml

```diff
@@ -2,5 +2,6 @@
 members = [
     "olpc-cjson",
     "tough",
+    "tough-ssm",
```

Comment:
@zmrow and @iliana, what do we think are the pros and cons of a subcrate vs. a feature?

Reply:
I don't think they are mutually exclusive. Having a subcrate could make the feature cleaner by excluding the entire subcrate.

Reply:
Sure, but maybe only cleaner for the developer and not the user?  The user has to add another dependency and use a different namespace to refer to that key source, when I think often that's accomplished with a feature.

Reply:
The strategy behind the trait is to make it easier for developers to use their own providers of keys. Implementing the SSM crate provides a good reference and example for developers and means `tough` isn't tied to any AWS-specific implementation, feature or otherwise.